sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+/**
+ *
+ *  CS3283/4 storage.js
+ *  This class acts as a facade for other developers to access to the database.
+ *  The implementation is a non-SQL local storage to support the WebApp.
+ *
+ */
+
+import constants from './constants';
+
+import entitiesCreator from './create/entitiesCreator';
+
+import query from './retrieve/query';
+
+import dataModifier from './data-modifier/dataModifier';
+
+import exceptions from './exceptions';
+
+let databaseName = constants.INDEXEDDB_NAME;
+
+// dummy function to initialize createTables promises
+function initCreateTable() {
+  if (typeof Promise === 'function') {
```

Comment:
You may not need to check for the existence of `Promise` in every function. 
Because if `Promise` is available in the first time, most likely it will be available all the time.

This check could be done the first time when other developer use/import your library.

It is however, not wrong to check for the existence of Promise in every function. 
I just felt that the checking could be too aggressive.
I will let you decide to remove these checks or keep them here.